test/uk/gov/hmrc/cdsimportsddsfrontend/controllers/DeclarationControllerSpec.scala

```diff
@@ -215,15 +298,39 @@ class DeclarationControllerSpec extends CdsImportsSpec
         "documentationType.previousDocGoodsItemId" -> Seq("1"),
         "documentationType.additionalInfoCode" -> Seq("00500"),
         "documentationType.additionalInfoDescription" -> Seq("IMPORTER"),
-        "documentationType.additionalDocCategoryCode" -> Seq("N"),
-        "documentationType.additionalDocTypeCode" -> Seq("935"),
-        "documentationType.additionalDocId" -> Seq("12345/30.09.2019"),
-        "documentationType.additionalDocLPCO" -> Seq("AC"),
-        "documentationType.additionalDocName" -> Seq("DocumentName"),
+        "documentationType.additionalDocument[0].categoryCode" -> Seq("N"),
```

Comment:
need to assert that the posted form data gets mapped to the expected XML elements

Reply:
In the section 4 branch we've created unit tests for the XML "service" you could follow that pattern